core/src/test/java/org/elasticsearch/index/query/FuzzyQueryBuilderTest.java

```diff
@@ -0,0 +1,145 @@
+                text = randomIntBetween(0, 10);
+            } else if (fieldName.equals(DOUBLE_FIELD_NAME)) {
+                text = randomDouble() * 10;
+            } else if (fieldName.equals(DATE_FIELD_NAME)) {
+                text = new DateTime(System.currentTimeMillis() - randomIntBetween(0, 1000000), DateTimeZone.UTC);
+            }
+            query = new FuzzyQueryBuilder(fieldName, text);
+        } else {
+            query = new FuzzyQueryBuilder(randomAsciiOfLengthBetween(1, 10), randomAsciiOfLengthBetween(1, 50));
+        }
+        if (randomBoolean()) {
+            Fuzziness fuzziness;
+            switch (randomIntBetween(0, 2)) {
+                case 0:
+                    fuzziness = Fuzziness.fromEdits(randomIntBetween(0, 2));
+                    break;
+                case 1:
+                    fuzziness = Fuzziness.fromSimilarity(randomFloat());
+                    break;
+                default:
+                    fuzziness = Fuzziness.AUTO;
+            }
+            query.fuzziness(fuzziness);
+        }
+        if (randomBoolean()) {
+            query.prefixLength(randomIntBetween(0, 10));
+        }
+        if (randomBoolean()) {
+            query.maxExpansions(randomIntBetween(1, 10));
+        }
+        if (randomBoolean()) {
+            query.transpositions(randomBoolean());
+        }
+        if (randomBoolean()) {
+            query.rewrite(randomFrom(REWRITE_METHOD));
+        }
+        if (randomBoolean()) {
+            query.boost(2.0f / randomIntBetween(1, 20));
+        }
+        if (randomBoolean()) {
+            query.queryName(randomAsciiOfLengthBetween(1, 10));
+        }
+        return query;
+    }
+
+    @Override
+    protected Query createExpectedQuery(FuzzyQueryBuilder queryBuilder, QueryParseContext context) throws IOException {
+        String fieldName = queryBuilder.fieldName();
+        String queryText = queryBuilder.text().toString();
+        
+        Query expectedQuery = null;
+        MappedFieldType fieldType = context.fieldMapper(fieldName);
+        if (fieldType != null) {
+            expectedQuery = fieldType.fuzzyQuery(queryText, queryBuilder.fuzziness(), queryBuilder.prefixLength(), 
+                    queryBuilder.maxExpansions(), queryBuilder.transpositions());
+        }
+        if (expectedQuery == null) {
+            expectedQuery = new FuzzyQuery(new Term(fieldName, queryText), queryBuilder.fuzziness().asDistance(queryText), 
+                    queryBuilder.prefixLength(), queryBuilder.maxExpansions(), queryBuilder.transpositions());
+        }
+        if (expectedQuery instanceof MultiTermQuery) {
+            MultiTermQuery.RewriteMethod rewriteMethod = FuzzyQueryBuilder.parseRewriteString(context, queryBuilder.rewrite());
+            QueryParsers.setRewriteMethod((MultiTermQuery) expectedQuery, rewriteMethod);
+        }
+        expectedQuery.setBoost(queryBuilder.boost());
+
+        if (queryBuilder.queryName() != null) {
```

Comment:
I think we shouldn't add the expectedQuery to the namedQuery in the context here, that should be done in toQuery() and it's part of what's beeing tested later.